include/irr/asset/CTexturePacker.h

```diff
@@ -145,16 +119,12 @@ class ITexturePacker
             (*offsets)[i] = (addr==CSquareAddressAllocator<uint32_t>::invalid_address) ? 
                 STexOffset::invalid_addr :
                 STexOffset(m_addrAlctr.unpackAddress_x(addr), m_addrAlctr.unpackAddress_y(addr), m_addrAlctr.unpackAddress_layer(addr));
-
-            //i didnt know where to put page table page table and filling code so it is here for now
-            uint32_t* const pgTab = reinterpret_cast<uint32_t*>(
-                reinterpret_cast<uint8_t*>(m_pageTable->getBuffer()->getPointer()) + m_pageTable->getRegions().begin()[i].bufferOffset
-                );
-            const uint32_t pitch = m_pageTable->getRegions().begin()[i].bufferRowLength;
-            const uint32_t offset = (pitch*(m_tilesPerDim>>i))*(*offsets)[i].layer() + (*offsets)[i].y()*pitch + (*offsets)[i].x();
-            for (uint32_t x = 0u; x < pgSqrSz; ++x)
-                for (uint32_t y = 0u; y < pgSqrSz; ++y)
-                    pgTab[offset + y*pitch + x] = (*offsets)[i].addr;
+        }
+        if (levelCount > _subres.levelCount)
+        {
+            const uint32_t addr = m_addrAlctr.alloc_addr(1u);//1 page for mip-tail (square 1x1)
+            if (addr!=CSquareAddressAllocator<uint32_t>::invalid_address)
+                (*offsets).back().addr |= STexOffset(m_addrAlctr.unpackAddress_x(addr), m_addrAlctr.unpackAddress_y(addr), m_addrAlctr.unpackAddress_layer(addr)).addr<<16;
```

Comment:
it doesn't get allocated in a different place in VT atlas

Reply:
? i dont understand

Reply:
the last page for the mip-tail it does not need a separate offset in the Virtual Address space